lib/internal/http2/core.js

```diff
@@ -540,15 +531,27 @@ function onSessionInternalError(code) {
     this[kOwner].destroy(new NghttpError(code));
 }
 
+function settingsCallback(cb, ack, duration) {
+  this[kState].pendingAck--;
+  if (typeof cb === 'function') {
+    const err = ack ? null : new errors.Error('ERR_HTTP2_SETTINGS_CANCEL');
+    cb(err, this.localSettings, duration);
+  }
+  if (ack)
+    this.emit('localSettings', this.localSettings);
```

Comment:
I would at least so a debug if ack is null.

Reply:
Just reworked the entire function a bit to clean it up more and add the debug output.